.github/workflows/run.yml

```diff
@@ -1,102 +1,113 @@
+        required: false
+        type: string
+        default: latest
+      earthly_version:
+        description: The version of Earthly to use.
+        required: false
+        type: string
+        default: latest
+    secrets:
+      dockerhub_username:
+        description: The token to use for logging into the DockerHub registry.
+        required: false
+      dockerhub_token:
+        description: The token to use for logging into the DockerHub registry.
+        required: false
+      earthly_runner_address:
+        description: |
+          The address of the Earthly runner that will be used to build the
+          Earthly files.
+        required: false
+      earthly_runner_secret:
+        description: |
+          The ID of the AWS secret holding Earthly remote runner credentials.
+          This secret must contain the runner address and the necessary TLS
+          certificates required to authenticate with it. If omitted, a remote
+          Earthly runner will not be configured.
+        required: false
 
 jobs:
   discover:
     runs-on: ubuntu-latest
     outputs:
       json: ${{ steps.check.outputs.json }}
+      paths: ${{ steps.check.outputs.paths }}
     steps:
       - uses: actions/checkout@v3
       - name: Setup CI
-        uses: input-output-hk/catalyst-ci/actions/setup@master
+        uses: input-output-hk/catalyst-ci/actions/setup@feat/targets-scanner
         with:
           cli_version: ${{ inputs.ci_cli_version }}
           configure_registries: "false"
           earthly_skip_install: "true"
       - name: Discover Earthly files
-        uses: input-output-hk/catalyst-ci/actions/discover@master
+        uses: input-output-hk/catalyst-ci/actions/discover@feat/targets-scanner
         id: discover
         with:
           targets: ${{ inputs.target }}
       - name: Check for empty output
         id: check
         run: |
-          output=$(echo '${{ steps.discover.outputs.json }}' | jq -rc)
+          json=$(echo '${{ steps.discover.outputs.json }}' | jq -rc)
+          paths=$(echo '${{ steps.discover.outputs.paths }}' | jq -rc)
           if [ "$output" == "null" ]; then
             echo "json=[]" >> $GITHUB_OUTPUT
+            echo "paths=[]" >> $GITHUB_OUTPUT
           else
-            echo "json=$output" >> $GITHUB_OUTPUT
+            echo "json=$json" >> $GITHUB_OUTPUT
+            echo "paths=$paths" >> $GITHUB_OUTPUT
           fi
 
   run:
     runs-on: ubuntu-latest
     needs: [discover]
-    if: needs.discover.outputs.json != '[]'
+    if: needs.discover.outputs.paths != '[]'
     strategy:
       fail-fast: false
       matrix:
-        earthfile: ${{ fromJson(needs.discover.outputs.json) }}
+        earthfile: ${{ fromJson(needs.discover.outputs.paths) }}
     steps:
+      - name: Get filtered targets
+        id: get_target
+        run: |
+          targets=$(echo '${{ needs.discover.outputs.json }}' | jq -r --arg key '${{ matrix.earthfile }}' '.[$key][]')
+          echo $targets
```

Comment:
We can keep this here, but we should make it a bit cleaner. Something like: `echo "Found targets: $targets"`